stable_baselines/common/vec_env/subproc_vec_env.py

```diff
@@ -161,3 +164,27 @@ def set_attr(self, attr_name, value, indices=None):
         for remote in [self.remotes[i] for i in indices]:
             remote.send(('set_attr', (attr_name, value)))
         return [remote.recv() for remote in [self.remotes[i] for i in indices]]
+
+
+def _flatten_obs(obs, space):
+    """
+    Flatten observations, depending on the observation space.
+
+    :param obs: (dict<ndarray>, tuple<ndarray> or ndarray) an observation.
+                Either a numpy array or a dict or tuple of numpy arrays.
+    :return (OrderedDict<ndarray>, tuple<ndarray> or ndarray) a flattened numpy array
+            or an OrderedDict or tuple of flattened numpy arrays.
+    """
+    assert isinstance(obs, list) or isinstance(obs, tuple)
+    assert len(obs) > 0
+
+    if isinstance(space, gym.spaces.Dict):
+        assert isinstance(obs[0], dict)
+        return collections.OrderedDict([(k, np.stack([o[k] for o in obs]))
+                                        for k in space.spaces.keys()])
```

Comment:
ok, this works because space.spaces.keys() will always be ordered. (Would not be True for python 3.5 if is was a normal dict)

Reply:
Yep, this is part of the interface for `gym.spaces.Dict`. I've added an assert statement to verify this and clarify it to the reader.